src/main/java/com/google/api/codegen/discovery/Schema.java

```diff
@@ -367,4 +368,30 @@ public boolean isPathParam() {
   public String toString() {
     return String.format("Schema \"%s\", type %s", getIdentifier(), type());
   }
+
+  /**
+   * @return hashCode that should be unique for each underlying Node in the Document. This function
+   *     includes the location of the node in its calculation, so two different nodes with the same
+   *     content but different parents will still have different hashCodes.
+   */
+  @Override
+  public int hashCode() {
```

Comment:
You need to either override both `hashCode` and `equals` or neither.

Reply:
Everybody's invited to the override party! Done.

Reply:
If I overwrite the hashCode function, then AutoValue doesn't generate a hashCode function. I wanted a custom hashCode function so that `parent` gets included as part of the hashCode, but AutoValue wouldn't have picked up `parent` because it isn't an abstract field getter.